examples/service_worker/src/lib.rs

```diff
@@ -134,213 +86,232 @@ struct PushSubscriptionKeys {
-                model.message = Some(format!("Service worker state changed to: {:?}", state))
-            }
-            ServiceWorkerStatus::SubscriptionRetrieved => {
-                model.message = Some("PushManager Subscription retrieved.".into())
-            }
-        },
-        Msg::RequestNotificationPermission => {
-            if let Some(sw_reg) = model.sw_reg.clone() {
-                orders.perform_cmd(async move {
-                    match register_push_manager(sw_reg.clone()).await {
-                        Ok(msg) => msg,
-                        Err(e) => Msg::FailedToRegisterPushManager(e),
-                    }
-                });
-            }
+            });
         }
-        Msg::ServiceWorkerActivated => {
-            model.activated = true;
-
-            let permission: web_sys::NotificationPermission = web_sys::Notification::permission();
-            if permission == web_sys::NotificationPermission::Granted {
-                model.notifications_granted = true;
-            }
+        Msg::NotificationPermissionRequested(Ok(_)) => {
+            let worker_error = ServiceWorkerError::InvalidPermissions;
+            model.error = Some(log_error(&worker_error));
         }
-        Msg::SubscriptionRetrieved(push_subscription) => {
-            log!(
-                "Got a push subscription of:",
-                serde_json::to_string_pretty(&push_subscription).unwrap()
-            );
-            model.notifications_granted = true;
-            model.push_subscription = Some(push_subscription);
-            orders.send_msg(Msg::StatusUpdate(
-                ServiceWorkerStatus::SubscriptionRetrieved,
-            ));
+        Msg::NotificationPermissionRequested(Err(worker_error)) => {
+            model.error = Some(log_error(&worker_error));
         }
-        Msg::SendMessage => {
-            web_sys::Notification::new("Hello from Seed service worker!")
-                .expect("Couldn't send notification.");
+        Msg::Subscribed(Ok(push_subscription)) => {
+            let worker_data = match &mut model.worker_data {
+                None => return,
+                Some(worker_data) => worker_data,
+            };
+
+            log!("subscription:", push_subscription);
+            worker_data.push_subscription = Some(push_subscription);
         }
-        Msg::SetServiceWorker(sw_reg) => {
-            model.sw_reg = Some(sw_reg);
+        Msg::Subscribed(Err(worker_error)) => {
+            model.error = Some(log_error(&worker_error));
         }
-        Msg::UnregisterServiceWorker => {
-            orders.perform_cmd(async move {
-                match unregister_service_worker().await {
-                    Ok(_) => Msg::UnregisteredServiceWorker,
-                    Err(e) => Msg::FailedToUnregisterServiceWorker(e),
-                }
+        Msg::SendMessage => {
+            // @TODO: It doesn't use push - it's only a simple local browser notification.
```

Comment:
We could always spin up a docker container that is running the service. If you're not opposed to it I'll set one up with a simple API. That way, we could include a button that when pressed, would send a fetch to the service. I was thinking that when the service receives the event, it would start a timer and push a notification every 15 seconds. You could then click a button to stop the push events.

Reply:
Examples with BE:
- https://github.com/seed-rs/seed/tree/master/examples/server_integration
   - It serves frontend directly from Actix.
- https://github.com/seed-rs/seed/tree/master/examples/websocket
   - You need to run `cargo make start_server websocker` to run the server, but it makes the code simpler.